### DISCUSSION TITLE
Endless loop of python-BaseException in debug

### DISCUSSION THREAD

**Discussion by IvanHahan** [miguelgrinberg/Flask-SocketIO#1600] [Q&A]
I've copied code snippet from docs how to run the socket io app:

```
from flask import Flask, render_template
from flask_socketio import SocketIO
app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret!'
app.config['DEBUG'] = True
socketio = SocketIO(app, async_mode='eventlet')

socketio.run(app)
```

But when I run it in prod, it's just stuck on run. Then I tried to run it in debug and it showed endless emitting of python-BaseException error:
![image](https://user-images.githubusercontent.com/17514015/115151005-a384ba80-a073-11eb-8473-51ca80f751f4.png)

flask-socketio==5.0.1
flask==1.1.2
eventlet==0.30.2

ubuntu: 20.04

**Comment by miguelgrinberg** [comment]
The code that you show does not come from the documentation, you've made some changes to it.

How do you run your application during development vs. production? That's missing in your description of the problem and is very likely the key to your problem.

**Comment by IvanHahan** [comment]
I found out this error is thrown when I import flask (not flask_socketio) in debug.
`from flask import Flask`

**Comment by miguelgrinberg** [comment]
I suggest you create a brand new virtual environment from scratch and reinstall all your dependencies. Also it wouldn't hurt to drop your IDE and test this in the terminal, to remove all unnecessary components from the problem.

**Comment by IvanHahan** [comment]
Yes, I did it. I was a bit wrong. It's working on ubuntu. It doesn't work on mac for some reason. Maybe I have some personal problem. It worked before, but now I can't get it to work even on the earlier flask versions. So maybe it's not flask related

**Comment by IvanHahan** [comment]
I've managed to reproduce the error on ubuntu machine.
I've updated anaconda, recreated environment.
python==3.9

My requirements.txt
pyyaml==5.4.1
flask==1.1.2
psycopg2-binary
flask-sqlalchemy==2.5.1
flask-socketio==5.0.1
flask-session==0.3.2
gunicorn
flasgger==0.9.5
numpy
eventlet==0.30.2

This happens when flask is imported IN DEBUG:
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException
python-BaseException

I also tried virtualenv instead of conda. The same error.

**Comment by miguelgrinberg** [comment]
Unfortunately there is nothing I can do to help with this, since you are not finding any problems with this project. I suggest you create a complete description of the issue, along with example code to help with reproducing it and send it to the appropriate project, which at this point I'm not sure if it is Flask, Python or Anaconda.
